include/mbedtls/config_psa.h

```diff
@@ -211,6 +265,38 @@ extern "C" {
 #define MBEDTLS_PSA_BUILTIN_ALG_TLS12_PSK_TO_MS         1
 #endif /* MBEDTLS_MD_C */
 
+#if defined(MBEDTLS_MD2_C)
+#define MBEDTLS_PSA_BUILTIN_ALG_MD2                     1
```

Comment:
Missing: `#define PSA_WANT_ALG_MD2`. See https://github.com/ARMmbed/mbedtls/pull/3912/commits/0b1ed6da82c95beca81b44b3a70212c8dbc4cb7e

It would help us if you could cherry-pick 0b1ed6da82c95beca81b44b3a70212c8dbc4cb7e and 02ec04c10f47aa9a1d77b68e9048619a48ddd320 and continue from there, to avoid conflicts between the same change done slightly differently.

Reply:
@gilles-peskine-arm So am I working on 3672 or is someone else doing that? I'm happy to finish 3672 here unless someone else is further along in the cycle. As for the other two commits here on other branches, do you just want me to include those changes since they are not merged into main development yet.

Reply:
I'm not working on #3672. I just needed this one easy part (the declarations in `config_psa.h`) for the RNG work.

Please don't include the other changes, just the two commits to `config_psa.h` (cherry-picked). I expect your PR will be merged first and then I can rebase mine on it without conflicts at this point.

Reply:
Sorry for the bother, but with both an Mbed TLS release and a Silabs deadline approaching, we can't afford to fully serialize changes at the moment.

Reply:
@gilles-peskine-arm Ok. I will incorporate those changes.

Reply:
@gilles-peskine-arm in 0b1ed6d there is defining of PSA_WANT_ in the section where MBEDTLS_PSA_CRYPTO_CONFIG is not defined. None of the other algorithms or features set PSA_WANT_ in this file, are these definitions an exception or was that something missed in the earlier reviews. We may want to consider updating that in a separate change set to keep that clean for the other features if it was missing the first time around.

Reply:
@gilles-peskine-arm I see the other changes in 02ec04c that makes sense that it would be everywhere. Is there a reason we need to define the PSA_WANT_ when MBEDTLS_PSA_CRYPTO_CONFIG is not defined? Seems like that feature is ensuring the generated file is not included so unsure why we would need any PSA_WANT definitions.

Reply:
@gilles-peskine-arm Updated and pushed changes, hopefully it is more in line with what you need.

Reply:
`PSA_WANT_xxx` needs to be defined in both configuration modes for the sake of code that uses `psa_crypto_xxx` and needs to know what cryptographic mechanisms are supported: application code, our TLS code, our unit tests, … See  https://github.com/ARMmbed/mbedtls/issues/3670#issuecomment-697966113, https://github.com/ARMmbed/mbedtls/pull/3697#discussion_r493888080, https://github.com/ARMmbed/mbedtls/blob/development/docs/proposed/psa-conditional-inclusion-c.md#user-content-old-style-definition-of-configuration-symbols.